app.py

```diff
@@ -14,35 +16,80 @@
 app_env = AppEnv()
 aws_rds = app_env.get_service(name="example-website-api-database")
 
-
-connection = psycopg2.connect(
-    host=aws_rds.credentials.get("host"),
-    user=aws_rds.credentials.get("username"),
-    password=aws_rds.credentials.get("password"),
-    database=aws_rds.credentials.get("name"),
-    port=aws_rds.credentials.get("port"),
-)
-
-
+# Set up pool as a function so in the case it dies we can recreate it 
+def setup_connection_pool():
+    
+    database_url = os.getenv("DATABASE_URL")
+    
+    pool = ConnectionPool(
+        conninfo=database_url,
+        min_size=1,
+        max_size=20,
+    )
+    return pool
+
+# Immediately create pool at module import 
+connection_pool = setup_connection_pool()
+
+# Maybe dont need atexit if using context manager per psycopg3 docs
+atexit.register(connection_pool.close)
+
+# Recreate pool if it dies
+def recreate_connection_pool():
+    global connection_pool
+    
+    # Close if it exists
+    if connection_pool is not None:
+        try:
+            connection_pool.close()
+        except Exception:
+            pass
+        
+    # Create new pool   
+    new_pool = setup_connection_pool()
+    
+    # Update variable
+    connection_pool = new_pool
+    
+    atexit.register(connection_pool.close)
+    
+    return connection_pool
 
 @app.route("/", methods=["GET"])
 def hello():
-    return "There is a table right behind this door!"
-
+    try:
+        return "There is a table right behind this door!"
+    except Exception as error:
+        return jsonify({"error": str(error)}), 500
 
 @app.route("/get_table", methods=["GET"])
-def get_table():
-
-    cursor = connection.cursor(cursor_factory=RealDictCursor)
-
-    query = "SELECT * FROM fdic_banks LIMIT 15"
-    cursor.execute(query)
-
-    rows = cursor.fetchall()
-    cursor.close()
-
-    return jsonify(rows)
-
-
+def get_table():   
```

Comment:
(non-actionable) This pattern in the psycopg3 docs is probably a bit better than try/except but does take extra time to check the connection: https://www.psycopg.org/psycopg3/docs/advanced/pool.html#connection-quality. Just leaving here as another idea